source/Cosmos.HAL2/BlockDevice/AtaPio.cs

```diff
@@ -250,20 +250,22 @@ public Status SendCmd(Cmd aCmd, bool aThrowOnError)
 			return xStatus;
 		}
 
-		protected string GetString(UInt16[] aBuffer, int aIndexStart, int aStringLength)
+		protected string GetString(ushort[] aBuffer, int aIndexStart, int aStringLength)
 		{
-			// Would be nice to convert to byte[] and use
-			// new string(ASCIIEncoding.ASCII.GetChars(xBytes));
-			// But it requires some code Cosmos doesnt support yet
-			var xChars = new char[aStringLength];
```

Comment:
Can you please keep the old array based approach and add the BitConverter fix to it?
I think it makes more sense to use an array, since we know the length of the string.